pkg/controller/certificaterequests/acme/acme.go

```diff
@@ -92,6 +93,19 @@ func (a *ACME) Sign(ctx context.Context, cr *v1alpha2.CertificateRequest, issuer
 		return nil, nil
 	}
 
+	// If the CommonName is also not present in the DNS names of the CSR then hard fail.
+	if len(csr.Subject.CommonName) > 0 && !util.Contains(csr.DNSNames, csr.Subject.CommonName) {
+		err = fmt.Errorf("%q does not exist in %s", csr.Subject.CommonName, csr.DNSNames)
+		message := fmt.Sprintf("Requested certificate contains CommonName not present in the requested DNS Subject Alternative Names: %s",
```

Comment:
Maybe note that ACME does not support setting the commonName field unless the domain is present in dnsNames? i.e. changes this message to:

```
The CSR PEM requests a commonName that is not present in the list of dnsNames. If a commonName is set, ACME requires that the value is also present in the list of dnsNames.
```

or something like that? I want to make sure users understand what they've done wrong, why it's wrong, and how to fix it based on this message.